### PR TITLE
log MoE aux/zloss into balance logs

### PR DESCRIPTION
**Pull request by huangjiyi**
## Summary
- add a shared `log_moe_losses` helper under the existing MoE balance logging gate
- record `aux_loss` and per-layer `aux_loss_layer_*` when the values are already computed
- record `zloss` and per-layer `zloss_layer_*` when the values are already computed
- keep the compute contract unchanged: this PR only logs existing values and does not force extra aux/z-loss computation

## Test plan
- `python -m py_compile src/paddlefleet/transformer/moe/moe_layer.py src/paddlefleet/transformer/moe/moe_utils.py`
- integration smoke verified from the erniebot fleet training path before publishing this PR

### CODE REVIEW

File: src/paddlefleet/transformer/moe/moe_utils.py [hunk 1]
```diff
@@ -255,6 +255,29 @@ def global_moe_balance_training_logs_enabled():
...
+
+    log = {}
+    if aux_loss is not None:
+        aux_loss = aux_loss.detach()
+        log["aux_loss"] = aux_loss
+        if layer_number is not None:
+            log[f"aux_loss_layer_{layer_number}"] = aux_loss
+
+    if z_loss is not None:
+        z_loss = z_loss.detach()
```
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/transformer/moe/moe_utils.py:273]: 保留这里的 `Tensor` 记录，不采纳这条建议。当前这条链路的下游日志容器可以处理 `paddle.Tensor`，并会在最终序列化时再做 `.item()`；另外如果这里改成 `float(aux_loss.item())`，会额外引入一次显式同步；这次 PR 的目标是只记录已经算出的值，不改变原有计算/同步契约。后续如果要统一日志值类型，建议单独在日志容器层做一致化，而不是只在这一条新路径上转成 Python float。

File: src/paddlefleet/transformer/moe/moe_utils.py [hunk 2]
```diff
@@ -255,6 +255,29 @@ def global_moe_balance_training_logs_enabled():
     return callable(is_enabled) and is_enabled()
 
 
+def log_moe_losses(layer_number, aux_loss=None, z_loss=None):
+    if not global_moe_balance_training_logs_enabled():
+        return
+    logs = get_global_training_logs()
+    if logs is None or not hasattr(logs, "update"):
```
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/transformer/moe/moe_utils.py:262]: 这条已采纳并补齐。当前 PR 里已经新增了 `log_moe_losses` 的单测，覆盖了 balance gate 开/关、`aux/z-loss is None`、以及 `MoELayer.forward()` 的调用点；另外按你的建议又补了一条 `layer_number=None` 时只写全局键、不写 layer 维度键的测试，已随最新 commit `e3200df` 推到这个 PR 分支。

File: src/paddlefleet/transformer/moe/moe_utils.py [hunk 3]
```diff
@@ -255,6 +255,29 @@ def global_moe_balance_training_logs_enabled():
...
+        aux_loss = aux_loss.detach()
+        log["aux_loss"] = aux_loss
+        if layer_number is not None:
+            log[f"aux_loss_layer_{layer_number}"] = aux_loss
+
+    if z_loss is not None:
+        z_loss = z_loss.detach()
+        log["zloss"] = z_loss
+        if layer_number is not None:
+            log[f"zloss_layer_{layer_number}"] = z_loss
```
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/transformer/moe/moe_utils.py:276]: 这里和 `aux_loss` 的判断一致，保留 `Tensor` 记录，不采纳这条建议。当前下游日志容器支持 `paddle.Tensor`，并在最终日志序列化时再转换；直接在这里做 `.item()` 会引入额外同步，而这次 PR 的约束是只透传已经算出的 `z_loss`，不额外改变运行时行为。

File: src/paddlefleet/transformer/moe/moe_layer.py [hunk 4]
```diff
@@ -744,6 +745,8 @@ def forward(self, hidden_states: paddle.Tensor) -> paddle.Tensor:
 
         _log_moe_md5(gates_masked, "gates_masked", layer_idx)
         _log_moe_md5(mask, "routing_mask", layer_idx)
+        if framework._dygraph_tracer()._has_grad:
+            log_moe_losses(layer_idx, aux_loss=aux_loss, z_loss=z_loss)
```
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/transformer/moe/moe_layer.py:749]: 这里有意记录的是原始的 router balance loss，而不是乘上 `router_aux_loss_coef` 之后注入训练主 loss 的缩放值，因此这条不采纳。coef 只用于训练时构造 router loss 或 `AddAuxiliaryLoss`。如果把 `aux_loss` 改成缩放后值，会把“模型当前均衡性”与“配置里选了多大的系数”混在一起，跨配置对比会失真。若后续确实需要观测训练实际注入强度，更合适的是单独增加一个 scaled 指标，而不是重定义 `aux_loss` 的含义。
